File: Lab7/Question5/src/Ostrich.java
public class Ostrich extends Bird {
	public Ostrich(String name) {
		super(name, 10);
	}

	@Override
	public boolean canFly() {
		return false;
	}

	public static void testOstrich() {
		Ostrich ostrich = new Ostrich("Minami");
		System.out.println(ostrich.getName() == "Minami");
		System.out.println(ostrich.getLegs() == 2);
		System.out.println(ostrich.getNumOfEggs() == 10);
		System.out.println(ostrich.canFly() == false);
	}
}

File: Lab7/Ostrich.kt
class Ostrich : Bird {
    constructor(name: String) : super(name, 10)

    override fun canFly(): Boolean {
        return false
    }

    override fun isDangerous(): Boolean {
        return true
    }

    companion object {
        fun testOstrich() {
            val ostrich = Ostrich("Minami")
            println(ostrich.getName() == "Minami")
            println(ostrich.canFly() == false)
            println(ostrich.getLegs() == 2)
            println(ostrich.isDangerous() == true)
            println(ostrich.getNumOfEggs() == 10)
            val ost: Flyer = Ostrich("ost")
            println(ost.getName() == "ost")
            println(ost.canFly() == false)
            println(ost.isDangerous() == true)
        }
    }
}